Assignment4/DCY1117-a180194/task09.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+"""Task09.ipynb
+
+Automatically generated by Colaboratory.
+
+Original file is located at
+    https://colab.research.google.com/github/DCY1117/Curso2022-2023-ODKG/blob/master/Assignment4/course_materials/notebooks/Task09.ipynb
+
+**Task 09: Data linking**
+"""
+
+!pip install rdflib
+github_storage = "https://raw.githubusercontent.com/FacultadInformatica-LinkedData/Curso2021-2022/master/Assignment4/course_materials/"
+
+from rdflib import Graph, Namespace, Literal, URIRef
+g1 = Graph()
+g2 = Graph()
+g3 = Graph()
+g1.parse(github_storage+"rdf/data03.rdf", format="xml")
+g2.parse(github_storage+"rdf/data04.rdf", format="xml")
+
+"""Busca individuos en los dos grafos y enlázalos mediante la propiedad OWL:sameAs, inserta estas coincidencias en g3. Consideramos dos individuos iguales si tienen el mismo apodo y nombre de familia. Ten en cuenta que las URI no tienen por qué ser iguales para un mismo individuo en los dos grafos."""
+
+from rdflib.plugins.sparql import prepareQuery
+
+from rdflib.namespace import RDF, RDFS, OWL
+
+VCARD = Namespace("http://www.w3.org/2001/vcard-rdf/3.0#")
+
+q1 = prepareQuery('''
+  SELECT ?Subject ?given ?family WHERE { 
+    ?Subject VCARD:Given ?given.
+    OPTIONAL{ ?Subject VCARD:Family ?family } 
+  } 
+  ''',
+  initNs = { "VCARD": VCARD }
+)
+
+print("Individuos, apodo y nombre de familia grafo 1##############")
+for r in g1.query(q1):
+  print(r)
+
+print("Individuos, apodo y nombre de familia grafo 2##############")
+for r in g2.query(q1):
+  print(r)
+
+print("inserta estas coincidencias en g3########")
+g3.add((URIRef('http://data.three.org#SaraJones'), OWL.sameAs, URIRef('http://data.four.org#0001')))
+g3.add((URIRef('http://data.three.org#JohnDoe'), OWL.sameAs, URIRef('http://data.four.org#0005')))
+
```

Comment:
It should be done programatically, not by hand

Reply:
Thanks, changes done.